lib/matplotlib/tests/test_backend_macos_common.py

```diff
@@ -0,0 +1,37 @@
+import sys
+import pytest
+import matplotlib.backends.backend_macos_common as bmc
+from matplotlib import cbook
+
+
+class NoString:
+    def __repr__(self):
+        pass
+
+
+if sys.platform == "darwin":
+    def test_set_mac_icon():
+        with pytest.raises(TypeError):
+            bmc.set_mac_icon()
+
+        with pytest.raises(RuntimeError):
```

Comment:
Not sure why these don't raise TypeError (nor whether we really need to test them).

Reply:
So the thing is, `matplotlib.cbook._get_data_path` returns a `PosixPath` object. It’s natural for people to pass in something that represents the path but isn’t a string. I can convert whatever argument is given me into a string, but that converts everything into a string, including these types. I could check its type to avoid that, but that’s not very Pythonic. Sort of a catch-22. Not sure of a good solution...

Reply:
My point is that if someone passes you something of the wrong type, it makes sense to raise a TypeError, not a RuntimeError.

Reply:
Yeah, absolutely, I want to do that. I just can’t think of a way to handle arbitrary types like `PosixPath` and also provide a TypeError. There’s nothing I know of that will only convert the various path types in Python while also erroring on things that aren’t paths.

Any ideas? Maybe I should give up trying to handle arbitrary types and only allow strings?

Reply:
see https://github.com/matplotlib/matplotlib/pull/14930#discussion_r309894455